### ISSUE TITLE
bigip_ssl_csr: Does not support relative paths in dest parameter

### ISSUE THREAD

**Issue by ncasperson** [F5Networks/f5-ansible#2345]
##### COMPONENT NAME

f5networks.f5_bigip
bigip_ssl_csr https://clouddocs.f5.com/products/orchestration/ansible/devel/f5_bigip/modules_2_0/bigip_ssl_csr_module.html

##### ANSIBLE VERSION

```
$ ansible --version
ansible [core 2.14.2]
  config file = /net_home/username/work/f5/ansible.cfg
  configured module search path = ['/net_home/username/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.11/site-packages/ansible
  ansible collection location = /net_home/username/work/f5/collections
  executable location = /usr/bin/ansible
  python version = 3.11.2 (main, Jun  6 2023, 07:39:01) [GCC 8.5.0 20210514 (Red Hat 8.5.0-18)] (/usr/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True
```
##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     16.1.3.3
  Build       0.0.3
  Edition     Point Release 3
  Date        Thu Dec 22 12:07:59 PST 2022
```
##### OS / ENVIRONMENT

N/A

##### SUMMARY

In the `bigip_ssl_csr` module when using a relative path value for the `dest` parameter the object in the BIG-IP is created under `sys crypto csr` however ansible does not create the csr file and the task fails.  Relative paths work with other F5 ansible modules in the `dest` parameter.

##### STEPS TO REPRODUCE

```
- hosts: all
  collections:
    - f5networks.f5_bigip
  connection: httpapi

  vars:
    ansible_user: "admin"
    ansible_httpapi_password: "secret"
    ansible_httpapi_port: 443
    ansible_network_os: f5networks.f5_bigip.bigip
    ansible_httpapi_use_ssl: yes
    ansible_httpapi_validate_certs: no

  tasks:
    - name: Create an SSL csr
      bigip_ssl_csr:
        name: csr-name
        key_name: /Common/existing.key
        common_name: csr-name
        #dest: /tmp/csr-name   # this works
        dest: folder/csr-name  # this fails
        state: present
```

##### EXPECTED RESULTS

```
[user.name@ansible f5]$ ls -la folder/
[user.name@ansible f5]$
[user.name@ansible f5]$ ansible-playbook --inventory ./inventories/inventory.ini testplaybook.yml -vvv
PLAYBOOK: testplaybook.yml *******************************************************************
1 plays in testplaybook.yml

PLAY [bigip] *************************************************************************************
TASK [Create an SSL csr] ************************************************************************
task path: /net_home/user.name/work/f5/testplaybook.yml:16
redirecting (type: connection) ansible.builtin.httpapi to ansible.netcommon.httpapi
changed: [bigip] => changed=true
  common_name: csr-name
  invocation:
    module_args:
      challenge_password: null
      city: null
      common_name: csr-name
      country: null
      dest: /tmp/csr-name
      email_address: null
      key_name: /Common/existing.key
      name: csr-name
      organization: null
      ou: null
      partition: Common
      province: null
      state: present

PLAY RECAP ********************************************************************************
bigip : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

[user.name@ansible f5]$ ls folder
[user.name@ansible f5]$ ls /tmp/csr-name
/tmp/csr-name
[user.name@ansible f5]$ 
```
##### ACTUAL RESULTS

```
TASK [Create an SSL csr] ***********************************************************
task path: /net_home/user.name/work/f5/testplaybook.yml:16
redirecting (type: connection) ansible.builtin.httpapi to ansible.netcommon.httpapi
fatal: [bigip]: FAILED! => changed=false
  module_stderr: '[Errno 2] No such file or directory: ''folder/csr-name'''
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error

PLAY RECAP ***********************************************************************
bigip : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
[user.name@ansible f5]$ ls folder/
[user.name@ansible f5]$

```

**Comment by pgouband** [comment]
Hi @ncasperson,

Have you tried the bigip_ucs_fetch module from the declarative collection?
https://clouddocs.f5.com/products/orchestration/ansible/devel/f5_bigip/modules_2_0/bigip_ucs_fetch_module.html#bigip-ucs-fetch-module-2

Timeout is by default 150. The accepted value range is between 150 and 1800 seconds.

**Comment by pgouband** [comment]
Hi @ncasperson,

Just to confirm did you used the module from the imperative collection or the declarative collection?
Imperative: https://clouddocs.f5.com/products/orchestration/ansible/devel/modules/bigip_ssl_csr_module.html#bigip-ssl-csr-module-3
Declarative: https://clouddocs.f5.com/products/orchestration/ansible/devel/f5_bigip/modules_2_0/bigip_ssl_csr_module.html#bigip-ssl-csr-module-2

This repo is for imperative collection.
If there is an issue with the declarative collection, the repo is https://github.com/F5Networks/f5-ansible-bigip

**Comment by f5killer** [comment]
This is not the real answer. I am looking below variable

vars:
    f5_server: "{{ your_f5_server }}"
    f5_username: "{{ your_f5_username }}"
    f5_password: "{{ your_f5_password }}"
    csr_name: "{{ your_csr_name }}"
    csr_common_name: "{{ your_csr_common_name }}"
    csr_subject_alternative_names:
      - "{{ your_csr_sans }}"
    csr_country: "{{ your_csr_country }}"
    csr_organization: "{{ your_csr_organization }}"
    csr_department: "{{ your_csr_department }}"
    csr_city: "{{ your_csr_city }}"
    csr_state: "{{ your_csr_state }}"
    csr_key_size: "{{ your_csr_key_size }}"
    csr_key_type: "{{ your_csr_key_type }}"
    csr_algorithm: "{{ your_csr_algorithm }}"

**Comment by pgouband** [comment]
Hi @f5killer,

I don't understand your answer. Why are you sharing variables?

The issue reported by @ncasperson is about a module from declarative collection (from the link provided).
This repository manages imperative collection so I'm asking if the issue about the path is on imperative or declarative collection.

**Comment by ncasperson** [comment]
> Hi @ncasperson,
> 
> Just to confirm did you used the module from the imperative collection or the declarative collection? Imperative: https://clouddocs.f5.com/products/orchestration/ansible/devel/modules/bigip_ssl_csr_module.html#bigip-ssl-csr-module-3 Declarative: https://clouddocs.f5.com/products/orchestration/ansible/devel/f5_bigip/modules_2_0/bigip_ssl_csr_module.html#bigip-ssl-csr-module-2
> 
> This repo is for imperative collection. If there is an issue with the declarative collection, the repo is https://github.com/F5Networks/f5-ansible-bigip

My apologies.  I am using the **declarative** collection.  I will submit my issue there.  Thank you for your help.
